core/src/avm2/globals/flash/display/bitmap_data.rs

```diff
@@ -144,7 +144,9 @@ pub fn init<'gc>(
     };
 
     new_bitmap_data.init_object2(activation.gc(), this);
-    this.init_bitmap_data(activation.gc(), new_bitmap_data);
+    if let Some(bitmap_data_obj) = this.as_bitmap_data_object() {
```

Comment:
Could this be an `.expect()` instead? It _should_ always be a BitmapDataObject here after all, right?